cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -456,9 +456,13 @@ public static KafkaCluster fromCrd(Kafka kafkaAssembly, KafkaVersion.Lookup vers
         result.setListeners(listeners);
 
         if (listeners != null) {
-            if (listeners.getPlain() != null
-                    && listeners.getPlain().getAuth() instanceof KafkaListenerAuthenticationTls) {
-                throw new InvalidResourceException("You cannot configure TLS authentication on a plain listener.");
+            if (listeners.getPlain() != null) {
+                if (listeners.getPlain().getAuth() instanceof KafkaListenerAuthenticationTls) {
+                    throw new InvalidResourceException("You cannot configure TLS authentication on a plain listener.");
+                }
+                if (listeners.getPlain().getAuth() instanceof KafkaListenerAuthenticationOAuth) {
```

Comment:
`else if`? It cannot be both, so you will save one evaluation of the expression?